internal/api/v1/user/contacts/contacts_api_test.go

```diff
@@ -213,23 +217,26 @@ func TestContactsAPI_ConfirmContact(t *testing.T) {
 			statusCode: http.StatusOK,
 			responder:  httpmock.NewJsonResponderOrPanic(http.StatusBadRequest, contactstest.NewBadRequestSPVError()),
 		},
-		fmt.Sprintf("HTTP POST /api/v1/contacts/%s/confirmation str response: 500", paymail): {
-			expectedErr: wallet.ErrUnrecognizedAPIResponse,
+		fmt.Sprintf("HTTP POST /api/v1/contacts/%s/confirmation str response: 500", contact.Paymail): {
+			expectedErr: errors.ErrUnrecognizedAPIResponse,
 			statusCode:  http.StatusInternalServerError,
 			responder:   httpmock.NewStringResponder(http.StatusInternalServerError, "unexpected internal server failure"),
 		},
 	}
 
-	URL := clienttest.TestAPIAddr + "/api/v1/contacts/" + paymail + "/confirmation"
+	url := clienttest.TestAPIAddr + "/api/v1/contacts/" + contact.Paymail + "/confirmation"
 
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
 			// when:
-			wallet, transport := clienttest.GivenSPVWalletClient(t)
-			transport.RegisterResponder(http.MethodPost, URL, tc.responder)
+			aliceClient, aliceTransport := clienttest.GivenSPVWalletClient(t)
+			aliceTransport.RegisterResponder(http.MethodPost, url, tc.responder)
+
+			passcode, err := aliceClient.GenerateTotpForContact(contact, 3600, 6)
+			require.NoError(t, err)
 
 			// then:
-			err := wallet.ConfirmContact(context.Background(), paymail)
+			err = aliceClient.ConfirmContact(context.Background(), contact, passcode, contact.Paymail, 3600, 6)
 			require.ErrorIs(t, err, tc.expectedErr)
```

Comment:
Hmm, I can't see the assertion for `statusCode` - even though it is in test definition.